graphite/client.go

```diff
@@ -68,7 +79,55 @@ func NewClient(carbon string, carbon_transport string, write_timeout time.Durati
 	}
 }
 
-func pathFromMetric(m model.Metric, prefix string) string {
+func loadContext(tmpl_context map[string]interface{}, m model.Metric) map[string]interface{} {
+	labels := make(map[string]string)
+	for ln, lv := range m {
+		labels[string(ln)] = string(lv)
+	}
+	tmpl_context["labels"] = labels
+	return tmpl_context
+}
+
+func match(m model.Metric, match map[model.LabelName]model.LabelValue, matchRE map[model.LabelName]config.Regexp) bool {
+	for ln, lv := range match {
+		if m[ln] != lv {
+			return false
+		}
+	}
+	for ln, r := range matchRE {
+		if !r.MatchString(string(m[ln])) {
+			return false
+		}
+	}
+	return true
+}
+
+func pathsFromMetric(m model.Metric, prefix string, rules []*config.Rule, tmpl_context map[string]interface{}) []string {
+	var paths []string
+
+	for _, rule := range rules {
+		match := match(m, rule.Match, rule.MatchRE)
+		if !match {
+			continue
+		}
+		if rule.Skip {
```

Comment:
// We have a rule to silence this metric
if rule.Continue == false && rule.Templ == nil {
  return nil
}